Mathlib/Data/Matrix/Composition.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2024 Yunzhou Xie. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Yunzhou Xie
+-/
+
+import Mathlib.Data.Matrix.Basic
+
+/-!
+## Main results
+# Composition of matrices
+This file shows that Mₙ(Mₘ(R)) ≃ Mₙₘ(R), Mₙ(Mₘ(R)) ≃ Mₘ(Mₙ(R)), Mn(Rᵒᵖ) ≃ₐ[K] Mₙ(R)ᵒᵖ
+and also different levels of equivalence when R is an AddCommMonoid,
+Semiring, and Algebra over a CommSemiring K.
```

Comment:
Can you follow the conventions laid out [here](https://leanprover-community.github.io/contribute/doc.html#header-comment) for the module docstring. In particular, start with a top-level header (not `##`), then a brief explanation of what the file does, then `## Main definitions` and document the main bijections you're constructing (see `Data/Matrix/Block.lean` for an example of what this should look like).

Reply:
Good point! done！